src/float32sr.jl

```diff
@@ -83,7 +83,6 @@ end
 # Promotion
 Base.promote_rule(::Type{Float16}, ::Type{Float32sr}) = Float32
 Base.promote_rule(::Type{Float64}, ::Type{Float32sr}) = Float64
```

Comment:
Can you elaborate more why this is needed? The reason I'm careful with promotions is that we want to make sure that Float16/32sr are actually used throughout an algorithm instead of being accidently converted
```julia
f(x) = 1.23*x
```
Could do two things: (i) Throw an error when called with `x::Float32sr` to highlight that your algorithm isn't type stable (what you throw in is what you get out) or (ii) promote to Float64 and have no stochastic rounding, which is probably not desired if one executes `f` with Float32sr.

So far I've opted for (i) because I wanted to highlight every type conversion that isn't explicit.

Having a promotion defined can easily lead to

Reply:
Related to the above discussion. To be clear, the promotion rules for Float16 and Float64 are already in the main SR.jl branch